scripts/core/editor3/helpers/insertAtomicBlockWithoutEmptyLinesAroundIt.js

```diff
@@ -0,0 +1,95 @@
+import {
+    BlockMapBuilder,
+    CharacterMetadata,
+    ContentBlock,
+    Modifier,
+    EditorState,
+    genKey
+} from 'draft-js';
+
+const Immutable = require('immutable');
+
+var List = Immutable.List,
+    Repeat = Immutable.Repeat;
+
+function getInsertionTarget(contentState, selectionState) {
+    var afterRemoval = Modifier.removeRange(contentState, selectionState, 'backward');
+    var targetSelection = afterRemoval.getSelectionAfter();
+    var afterSplit = Modifier.splitBlock(afterRemoval, targetSelection);
+
+    // return unsplit content if after the splitting first line is empty
+    if (afterSplit.getBlockForKey(afterSplit.getSelectionBefore().getAnchorKey()).getText().length < 1) {
+        return {
+            contentState: afterRemoval,
+            selectionState: targetSelection
+        };
+    }
+
+    var insertionTarget = afterSplit.getSelectionAfter();
+
+    return {
+        contentState: afterSplit,
+        selectionState: insertionTarget
+    };
+}
+
+/*
+    insertAtomicBlockWithoutEmptyLinesAroundIt: function(
+        editorState: EditorState,
+        entityKey: string,
+        character: string
+    ): EditorState
+*/
+
+function insertAtomicBlockWithoutEmptyLinesAroundIt(editorState, entityKey, character) {
```

Comment:
as much as I like descriptive names this seems a bit too much :wink:

Reply:
I wanted to make it clear it's not the same as the original `insertAtomicBlock` from `draft-js` helpers I based this one on. Do you suggest to keep the name the same?

Reply:
I think `insertAtomicBlockWithoutEmptyLines` would be enough